arangod/IResearch/IResearchDataStore.cpp

```diff
@@ -1037,9 +1078,15 @@ Result IResearchDataStore::commitUnsafeImpl(
       }
     }
 #endif
-    // get new reader
-    auto reader = _dataStore._reader.reopen(_format);
 
+#ifdef USE_ENTERPRISE
+    auto reader = forceOpen() ? irs::directory_reader::open(
+                                    *(_dataStore._directory), _format,
+                                    _dataStore._readerOptions)
+                              : _dataStore._reader.reopen(_format);
```

Comment:
same here: how does the reader know about the value of `_useSearchCache`?

Reply:
Reference to _useSearchCache is stored in reader options. Implicitly - as lambda capture.